crypto/bio/bss_conn.c

```diff
@@ -188,7 +188,16 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
 
         case BIO_CONN_S_BLOCKED_CONNECT:
             i = BIO_sock_error(b->num);
-            if (i) {
+            if (i != 0) {
+                if ((c->addr_iter = BIO_ADDRINFO_next(c->addr_iter)) != NULL) {
+                    /*
+                     * if there are more addresses to try, do that first
+                     */
+                    BIO_closesocket(b->num);
+                    c->state = BIO_CONN_S_CREATE_SOCKET;
+                    ERR_clear_error();
+                    break;
+                }
                 BIO_clear_retry_flags(b);
```

Comment:
Okay, I'd still move this BIO_clear_retry_flags up before the if,
since the previous connect is finished unsucessfully now.
(at least unless that does not work for any reason)

Reply:
Done